src/include/storage/recovery/recovery_manager.h

```diff
@@ -85,6 +103,10 @@ class RecoveryManager : public common::DedicatedThreadOwner {
         catalog::postgres::Builder::GetConstraintTableSchema();
     catalog_table_schemas_[catalog::postgres::INDEX_TABLE_OID] = catalog::postgres::Builder::GetIndexTableSchema();
     catalog_table_schemas_[catalog::postgres::TYPE_TABLE_OID] = catalog::postgres::Builder::GetTypeTableSchema();
+
+    if (enable_metric_) {
+      metrics_manager_ = common::ManagedPointer(new metrics::MetricsManager);
```

Comment:
The `RecoveryManager` should not instantiate a new `MetricsManager`, it should be passed in as a managed pointer to the `RecoveryManager`. Your test should instantiate a `MetricsThread`, and you should pass in the `MetricsThread`'s `MetricsManager`. 

If metrics is disabled, pass in `DISABLED` instead. This will also remove the need for the `enable_metric` flag, just instead check if `metrics_manager_ != DISABLED`

Reply:
No longer needed as I am using the metrics thread, which does the registering for us